octobot/src/server/github_handler.rs

```diff
@@ -406,6 +407,25 @@ impl GithubEventHandler {
         let mut participants = pull_request.assignees();
         // add the author of the PR
         participants.push(pull_request.user().clone());
+        // add team members
+        let teams = pull_request.teams();
+        let futures = teams.iter().map(|t| {
+            self.github_session
+                .get_team_members(t.org(), t.slug.as_str())
+        });
+        let teams = join_all(futures).await;
+        let mut members: Vec<User> = vec![];
+        teams.iter().for_each(|r| {
+            let member = match r {
+                Ok(r) => r.clone(),
+                Err(e) => {
+                    error!("Error getting team members: {}", e);
+                    vec![]
+                }
+            };
+            members.append(&mut member.clone());
+        });
+        participants.append(&mut members);
```

Comment:
here is a fun entry into rust ownership / borrowing. :) 

here is an alternate way of writing this:

```suggestion
        let team_members: Vec<github::User> = teams
            .into_iter()
            .filter_map(|r| match r {
                Ok(r) => Some(r),
                Err(e) => {
                    error!("Error getting team members: {}", e);
                    None
                }
            })
            .flatten()
            .collect::<Vec<github::User>>();
        participants.extend(team_members.into_iter());
```

or even shorter:
```suggestion
        participants.extend(
            teams
                .into_iter()
                .filter_map(|r| match r {
                    Ok(r) => Some(r),
                    Err(e) => {
                        error!("Error getting team members: {}", e);
                        None
                    }
                })
                .flatten(),
        );
```

thoughts:
- i like using the functional paradigms of `filter` and `map` (and here: in one go: `filter_map`)
- you can use `into_iter` to take ownership of the elements you're iterating over so you don't have to clone them

Reply:
Perfect. That's exactly what I wanted to do. 😅